src/main/java/com/minecolonies/coremod/client/gui/WindowTownHall.java

```diff
@@ -635,6 +635,23 @@ private void createAndSetStatistics()
                 final String job = splitString[length - 1].toLowerCase(Locale.ENGLISH);
                 jobCountMap.put(job, jobCountMap.get(job) == null ? 1 : (jobCountMap.get(job) + 1));
             }
+		}
+
+        final Map<String, Integer> jobMaxCountMap = new HashMap<>();
+        for (@NotNull final IBuildingView building : townHall.getColony().getBuildings())
+        {
+            if (!building.isBuilding())
+            {
+                String buildingName = building.getSchematicName().toLowerCase();
+                if (!buildingName.equals("warehouse") && !buildingName.equals("barracks") && !buildingName.equals("citizen") && !buildingName.equals("townhall"))
```

Comment:
I'd much rather here you do something akin to `building instanceOf BuildingWarehouse` or etc, i think string references in this case are a lesser alternative

Reply:
Yep!

Reply:
Would be BuildWarehouse.View probably in this case though (as a note)

Reply:
instanceof AbstractBuildingWorker.View might make this even simpler